var/spack/repos/builtin/packages/geant4/package.py

```diff
@@ -24,62 +24,80 @@
-        cmake_args = list(std_cmake_args)
-        cmake_args.append('-DXERCESC_ROOT_DIR:STRING=%s' %
-                          spec['xerces-c'].prefix)
-        cmake_args.append('-DGEANT4_BUILD_CXXSTD=c++14')
-
-        cmake_args += ['-DGEANT4_USE_GDML=ON',
-                       '-DGEANT4_USE_SYSTEM_EXPAT=ON',
-                       '-DGEANT4_USE_SYSTEM_ZLIB=ON',
-                       '-DGEANT4_USE_SYSTEM_CLHEP=ON']
-
-        # fixme: turn off data for now and maybe each data set should
-        # go into a separate package to cut down on disk usage between
-        # different code versions using the same data versions.
-        cmake_args.append('-DGEANT4_INSTALL_DATA=OFF')
-
-        # http://geant4.web.cern.ch/geant4/UserDocumentation/UsersGuides/InstallationGuide/html/ch02s03.html
-        # fixme: likely things that need addressing:
-        # -DGEANT4_USE_OPENGL_X11=ON
+    def build_type(self):
+        spec = self.spec
+        if '+debug' in spec:
+            return 'Debug'
+        else:
+            return 'Release'
+
+    def cmake_args(self):
+        spec = self.spec
+
+        options = [
+            '-DGEANT4_USE_GDML=ON',
+            '-DGEANT4_USE_SYSTEM_CLHEP=ON',
+            '-DGEANT4_USE_G3TOG4=ON',
+            '-DGEANT4_INSTALL_DATA=ON',
+            '-DGEANT4_BUILD_TLS_MODEL=global-dynamic',
+            '-DGEANT4_BUILD_MULTITHREADED=ON',
+            '-DGEANT4_USE_USOLIDS=ON',
+            '-DGEANT4_USE_SYSTEM_EXPAT=OFF',
+            '-DXERCESC_ROOT_DIR:STRING=%s' %
+            spec['xerces-c'].prefix,
+            '-DUSolids_DIR=%s' %
+            join_path(spec['vecgeom'].prefix, 'lib/CMake/USolids')]
+
+        arch = platform.system().lower()
+        if arch is not 'darwin':
+            options.append('-DGEANT4_USE_OPENGL_X11=ON')
+            options.append('-DGEANT4_USE_XM=ON')
+            options.append('-DGEANT4_USE_RAYTRACER_X11=ON')
+
+        if '+cxx11' in spec:
+            options.append('-DGEANT4_BUILD_CXXSTD=c++11')
+        if '+cxx14' or '+cxx1y' in spec:
+            options.append('-DGEANT4_BUILD_CXXSTD=c++14')
 
         if '+qt' in spec:
-            cmake_args.append('-DGEANT4_USE_QT=ON')
-
-        build_directory = join_path(self.stage.path, 'spack-build')
-        source_directory = self.stage.source_path
+            options.append('-DGEANT4_USE_QT=ON')
+            options.append(
+                '-DQT_QMAKE_EXECUTABLE=%s' %
+                spec['qt'].prefix + '/bin/qmake'
+            )
 
-        with working_dir(build_directory, create=True):
-            cmake(source_directory, *cmake_args)
-            make()
-            make("install")
+        return options
 
     def url_for_version(self, version):
         """Handle Geant4's unusual version string."""
-        return "http://geant4.cern.ch/support/source/geant4.%s.tar.gz" % version
+        return (
+            "http://geant4.cern.ch/support/source/geant4.%s.tar.gz" % version
+        )
```

Comment:
Just so you know, we exempt URLs from the flake8 like length check, so you never need to worry about them.